scripts/reportInstalls.ts

```diff
@@ -0,0 +1,160 @@
+const setupAppInsights = () => {
+  appInsights
+    .setup(DEFAULT_AIKEY)
+    .setAutoCollectRequests(false)
+    .setAutoCollectPerformance(false)
+    .setAutoCollectExceptions(false)
+    .setAutoCollectDependencies(false)
+    .setAutoDependencyCorrelation(false)
+    .setAutoCollectConsole(false)
+    .setUseDiskRetryCaching(true)
+    .start();
+  return appInsights.defaultClient;
+};
+
+const appInsightsClient = setupAppInsights();
+
+const reportInstallCount = (
+  extensionIdentifier: string,
+  extensionStats: TelemetryData
+) => {
+  const name = `${extensionIdentifier}/${eventName}`;
+  logger(`Reporting install count for ${extensionIdentifier}`, {
+    name,
+    properties: extensionStats
+  });
+
+  if (!DRY_RUN) {
+    appInsightsClient.trackEvent({
+      name,
+      properties: extensionStats
+    });
+  }
+};
+
+const reportVSCodeMarketplaceInstalls = async (extension: string) => {
+  logger(`Reporting installs for ${extension}`);
+  const command = `vsce show salesforce.${extension} --json`;
+  const stats = await promisifiedExec(command);
+  const jsonStats: { statistics: { statisticName: string; value: string }[] } =
+    JSON.parse(stats.stdout);
+  const extensionStats: InstallTelemetryDataMS = {
+    install: '',
+    averagerating: '',
+    ratingcount: '',
+    trendingdaily: '',
+    trendingmonthly: '',
+    trendingweekly: '',
+    updateCount: '',
+    weightedRating: '',
+    downloadCount: '',
+    marketplace: marketPlaces.MICROSOFT,
+    'common.extname': extension
+  };
+
+  jsonStats.statistics.forEach(statObj => {
+    extensionStats[statObj.statisticName] = String(statObj.value);
+  });
+
+  reportInstallCount(extension, extensionStats);
+};
+
+const reportOpenVSXInstalls = async (extension: string) => {
+  const command = `ovsx get --metadata salesforce.${extension}`;
+  const stats = await promisifiedExec(command);
+  const jsonStats: { downloadCount: number; reviewCount: number } = JSON.parse(
+    stats.stdout
+  );
+  const extensionStats: InstallTelemetryData = {
+    install: String(jsonStats.downloadCount),
+    ratingcount: String(jsonStats.reviewCount),
+    marketplace: marketPlaces.OPENVSX,
+    'common.extname': extension
+  };
+
+  reportInstallCount(extension, extensionStats);
+};
+
+const execute = async () => {
+  logger('Report Installs Script Running...');
+
+  for (const extension of extensionsToTrack) {
+    logger('Report VSCode Marketplace Installs...');
+    await reportVSCodeMarketplaceInstalls(extension);
+
+    logger('Report OpenVSX Installs...');
+    await reportOpenVSXInstalls(extension);
+  }
+};
+
+execute()
+  .then(() => {
+    logger('reportInstalls script completed successfully');
+    process.exit(0);
+  })
+  .catch(err => {
+    logger('Error running reportInstalls script', err);
+    process.exit(1);
+  });
```

Comment:
output includes all the stdout from the cli calls, so it's pretty verbose, but here's an example log

```
*** Reporting install count for salesforcedx-einstein-gpt {
  "name": "salesforcedx-einstein-gpt/installStats",
  "properties": {
    "install": "15312",
    "averagerating": "3.4000000953674316",
    "ratingcount": "5",
    "trendingdaily": "0.013083011709295479",
    "trendingmonthly": "12.827892980964217",
    "trendingweekly": "2.740890953097403",
    "updateCount": "99065",
    "weightedRating": "4.068803184661991",
    "downloadCount": "282",
    "marketplace": "MS"
  }
}
```